torch/distributed/_tensor/_xla.py

```diff
@@ -0,0 +1,200 @@
+            manager, default: None
+        placements (List[:class:`Placement`], optional): the placements that
+            describes how to place the tensor on DeviceMesh, must have the same
+            number of elements as `device_mesh.ndim`. If not specified, we will
+            by default replicate the tensor across the `device_mesh` from the
+            first rank of each dimension of the `device_mesh`.
+
+    Returns:
+        A :class:`XLAShardedTensor` object
+
+    .. note:: We return a XLAShardedTensor with a global view and access to local shards.
+    The successive ops would be programmed as if on a single-device and without calling
+    any explicit collective ops. The actual sharded computation on the sharding annotated tensor
+    happens lazily, is transparent to the user. In the future, we will introduce
+    a new DTensor type for this kind of programming-mode (single-controller) and return.
+    """
+    # device_mesh is not optional in xla_distribute_tensor
+    dt_mesh = device_mesh
+    assert dt_mesh.device_type == "xla"
+
+    # convert to XLA device mesh
+    xla_mesh = convert_to_xla_mesh(dt_mesh)
+    assert xla_mesh.mesh_shape == tuple(dt_mesh.mesh.size())
+
+    # convert tensor to the corresponding device type if it's not in that device type
+    if not tensor.is_meta:
+        tensor = tensor.to(dt_mesh.device_type)
+    # set default placements to replicated if not specified
+    if placements is None:
+        placements = [Replicate() for _ in range(dt_mesh.ndim)]
+    assert (
+        len(placements) == dt_mesh.ndim
+    ), "`placements` must have the same length as `device_mesh.ndim`! "
+    f"Found placements length: {len(placements)}, and device_mesh.ndim: {dt_mesh.ndim}."
+    # convert placements to xla partition spec
+    partition_spec = convert_to_xla_partition_spec(tensor, placements)
+    assert len(tensor.shape) == len(
+        partition_spec
+    ), "`partition_spec` from `placements` must have the same length as `tensor.length`! "
+    f"Found tensor shape length: {len(tensor.shape)}, and partition_spec length: {len(partition_spec)}."
+
+    global_tensor = tensor
+    if type(tensor).__name__ == "DTensor":
+        raise ValueError(
+            "Cannot distribute a DTensor with local tensor on xla devices."
+            "The input tensor must be global."
+        )
+    if type(tensor).__name__ == "XLAShardedTensor":
+        sharding_type = tensor.sharding_type  # type:ignore[attr-defined]
+        assert (
+            sharding_type is None or sharding_type == ShardingType.REPLICATED
+        ), "XLAShardedTensor `tensor` is already annotated with non-replication sharding. "
+        "Clear the existing sharding annotation first, by callling torch_xla.experimental.xla_sharding.clear_sharding API."
+        global_tensor = tensor.global_tensor  # type:ignore[attr-defined]
+    assert global_tensor is not None, "distributing a tensor should not be None"
```

Comment:
do we need a 
```
else:
  raise ValueError
```
here? Can we handle non XLASHardedTensor?

Reply:
We handle torch.Tensor or XLAShardedTensor (its global representation). We decided to block DTensor to be consistent with DTensor's eager api.